### DISCUSSION TITLE
Mobile Browser + Sockets will not re-connect when re-focused

### DISCUSSION THREAD

**Discussion by Kikketer** [supabase/supabase#18533] [Questions]
My basic goal is to have a homescreen style app with sockets enabled.  However when the user switches between apps the socket connection is lost (which is expected since it's no longer in focus and the OS should likely stop such communications).

So to mitigate the lost socket I've added a "visibility" handler that will re-establish the socket connection when the app is visible again.  Thing is the socket tends to throw an error and then timeout.  It is never able to re-establish the connection.  On desktop it works fine and there's no issue but on mobile browsers (Safari, Brave, etc) it'll throw a channel error then closed + timeout forever.  The only way I've found to fix it is to refresh the page, but I'd rather not do that to preserve the "app" feel.

A repo to reproduce the results: https://github.com/Kikketer/Supabase-Socket-POC/blob/main/src/App.tsx#L41
It's running on Vercel here: https://supabase-socket-poc.vercel.app/
If you open that site on your mobile device, then background your browser for a minute or two (at leas in iOS) then come back, you'll see the logs of the event not working:

**Mobile:**
![IMG_CAF83E8F5846-1](https://github.com/supabase/supabase/assets/958042/95e6ccf3-865e-43dd-b491-ad0f23f7ae1e)

**Desktop:**
![image](https://github.com/supabase/supabase/assets/958042/504f1963-2e80-41cc-b928-4bc5f3269f86)

Is there something else I should be doing to re-establish the connection on mobile without refreshing the page?  I've tried many things like `unsubscribe` and re-creating the client.

**Comment by GaryAustin1** [comment]
I've not looked at mobile in awhile with realtime but here: https://github.com/GaryAustin1/Realtime2 and the linked discussion at top and another users linked repository might be useful to you.  My linked discussion in that repository was testing alot on mobile back then.

**Reply by Kikketer** [reply]
Thank you, this discussion looks almost exactly like what I'm seeing: https://github.com/orgs/supabase/discussions/5641.  But the discussion is more focused on loss of data, which I'm actually not too concerned about with the docs and other things kind of saying "sockets aren't a guarantee" in terms of events never even firing.

With the mindset that I'm not going to use the sockets for mission critical stuff and just rely on the standard api style interactions to make sure the data is actually in-sync, I still have issues just re-establishing the connection.  Lost data or whatever, it just refuses to re-establish the connection after about 10-20 seconds in the background on mobile.

I guess I'm mostly raising the concern that the socket refuses to connect vs "lost some data while the connection was happening".  No matter what I do, except an actual page refresh, doesn't seem to get the socket re-established. If I followed the second pattern of only doing the SELECT after the socket connects, it would never happen in this specific mobile-only issue.

My alternatives:
1.  Ignore the failures and just rely on the standard API interactions.  The app i'm making isn't entirely socket-critical but was using it as somewhat of a POC but also "if a user edits this field while you are viewing it" edge case.
2. POC some alternate framework/technology-stack.  I really don't want to do this but tempted to at least see if this issue exists with other products.

**Reply by GaryAustin1** [reply]
I don't have a way to test on mobile right now.  That discussion involved mobile and my linking those was because I was using visibility or error to close the connection and restart on visible and it worked on mobile AT THAT TIME.
